packages/core-plugin-api/src/apis/system/useAnalytics.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AnalyticsApi, AnalyticsTracker } from '../definitions/AnalyticsApi';
+import { useExtensionAwareness } from '../../extensions';
+import { useApi, ApiRef } from './';
+
+/**
+ * Get a pre-configured analytics tracker.
+ */
+export function useAnalytics(apiRef: ApiRef<AnalyticsApi>): AnalyticsTracker {
```

Comment:
Why is that a parameter? Is there really a use case for passing in a custom ref? Feels like at that point you've left the basic safe analytics api haven and made your own right? 🤔

Reply:
I think I ended up with this because I was playing with things directly adjacent the `useApi` hook and importing the specific ref in that file seems to make Backstage explode.

You're right though, no need to take this as a param.

Reply:
✅